typer 0.28.0 breaks build on Windows due to quoted path

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Nature of Your Project

TypeScript

### Current Behavior

After updating cds-type to 0.28.0 and cds 8.x, executing `cds build` throws error:

```
error TS5033: Could not write file 'C:/Entwicklung/myproject/gen/srv"/srv/my-service.js': EINVAL: invalid argument, mkdir 'C:\Entwicklung\myproject\gen\srv"'.

    at Plugin.build (C:\Entwicklung\cap-ih\node_modules\@cap-js\cds-typer\cds-plugin.js:139:19)
    at async Promise.allSettled (index 0)
    at async BuildTaskEngine._executePipeline (C:\Users\daniel.boelts\AppData\Roaming\npm\node_modules\@sap\cds-dk\lib\build\buildTaskEngine.js:159:29)
    at async BuildTaskEngine._executeBuildTasks (C:\Users\daniel.boelts\AppData\Roaming\npm\node_modules\@sap\cds-dk\lib\build\buildTaskEngine.js:150:25)
    at async BuildTaskEngine.processTasks (C:\Users\daniel.boelts\AppData\Roaming\npm\node_modules\@sap\cds-dk\lib\build\buildTaskEngine.js:59:33)
    at async build (C:\Users\daniel.boelts\AppData\Roaming\npm\node_modules\@sap\cds-dk\lib\build\index.js:19:16)
    at async Object.build (C:\Users\daniel.boelts\AppData\Roaming\npm\node_modules\@sap\cds-dk\bin\build.js:171:5)
    at async Object.exec (C:\Users\daniel.boelts\AppData\Roaming\npm\node_modules\@sap\cds-dk\bin\cds.js:87:16)
```

### Expected Behavior

Does not throw error.

### Steps To Reproduce

Just run `cds build` on a CAP-Project. It worked with an erlier version of cds-typer.

### Environment

```markdown
- **OS**: Windows 11 (also tested on Windows 10)
- **Node**: 20.18.0
- **npm**: 10.8.2
- **cds-typer**: 0.28.0
- **cds**: 8.3.1
- **tsc**: 5.6.3
```


### Repository Containing a Minimal Reproducible Example

_No response_

### Anything else?

Commit https://github.com/cap-js/cds-typer/commit/c58b8b6c8f78ac1275aba6136dc5035ca92323ef now double-quotes the outDir of tsc command. This leads to this issue. Removing the double quotes makes it work again. I also tried to replace the double-quotes with single-quotes: same behavior. Without double quotes and spaces in path names the issue is like in #330. 

Running exactly the same command (with double-quotes) in cmd and powershell: everything works.

And now the interesting thing: the output of tsc shows unix/posix-style path separators. So i tried to convert the "this.task.dest" to posix-style separators (`this.task.dest.replaceAll("\\", "/")`): it works as expected. 

So it seems there is an incompatibility with Windows-Paths, quotes, the exec-method and the tsc command

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,10 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
-## Version 0.29.0 - TBD
+## Version 0.28.1 - TBD
+
+### Fixed
+- `cds build` no longer fails on Windows with an `EINVAL` error.
 
 ## Version 0.28.0 - 24-10-24
 ### Added

--- a/cds-plugin.js
+++ b/cds-plugin.js
@@ -110,7 +110,7 @@ cds.build?.register?.('typescript', class extends cds.build.Plugin {
         DEBUG?.('building without config')
         // this will include gen/ that was created by the nodejs task
         // _within_ the project directory. So we need to remove it afterwards.
-        await exec(`npx tsc --outDir "${this.task.dest}"`)
+        await exec(`npx tsc --outDir "${this.task.dest.replace(/\\/g, '/')}"`) // see https://github.com/cap-js/cds-typer/issues/374
         rmDirIfExists(path.join(this.task.dest, cds.env.build.target))
         rmDirIfExists(path.join(this.task.dest, this.#appFolder))
     }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@cap-js/cds-typer",
-  "version": "0.28.0",
+  "version": "0.28.1",
   "description": "Generates .ts files for a CDS model to receive code completion in VS Code",
   "main": "index.js",
   "repository": "github:cap-js/cds-typer",

